core/src/main/java/hudson/Functions.java

```diff
@@ -1143,11 +1144,14 @@ private static String normalizeURI(String uri) {
      *
      * @since 1.515
      * @param p the Item we want the relative display name
-     * @param g the ItemGroup used as point of reference for the item
+     * @param g the ItemGroup used as point of reference for the item.
+     *          If the group is not specified, item's path will be used.
      * @param useDisplayName if true, returns a display name, otherwise returns a name
      * @return
-     *      String like "foo » bar"
+     *      String like "foo » bar".
+     *      {@code null} if item is null or if one of its parents is not an {@link ItemGroup}.
      */
+    @Nullable
     public static String getRelativeNameFrom(Item p, ItemGroup g, boolean useDisplayName) {
```

Comment:
Really these methods should have been in `Items` (cf. #3148). Too late I suppose, unless we deprecate the `Functions` versions, which seems too intrusive.